### ISSUE TITLE
”ADS1220_WE“ usage

### ISSUE THREAD

**Issue by xiyangdiy** [wollewald/ADS1220_WE#1]
How to use "SPIClass *s" in ADS1220_WE(int cs, int drdy, SPIClass *s), I would be grateful if you could tell me

**Comment by wollewald** [comment]
Hi @xiyangdiy,

there two ways to create your ADS1220 object:
````
ADS1220_WE ads = ADS1220_WE(ADS1220_CS_PIN, ADS1220_DRDY_PIN);
// or:
ADS1220_WE ads = ADS1220_WE(ADS1220_CS_PIN, ADS1220_DRDY_PIN, &SPI);
````
The second option uses "SPIClass *s". If you work with just one SPI object, then it doesn't make a difference which option you take. The SPI object is passed as a reference (with the "&"). That means the original object is used. So it's not too interesting for MCUs which only have one SPI interface. But if you work for example with an ESP32 which has two SPI interfaces and you want to use both then you would need to pass the SPI objects.

**Comment by wollewald** [comment]
By the way thanks for the first star! And sorry for all the releases with the last weeks. Since it is new, I still found some bugs.

**Comment by xiyangdiy** [comment]
Thank you for your answer. Can this library use SoftSPI? If so, do you have a specific example? I found a library, its name is "SoftSPIB", it can customize spi pins, but I don't know how to use it with "ADS1220_WE" library.

**Comment by xiyangdiy** [comment]
My MCU is ESP32-S3.

**Comment by wollewald** [comment]
That would need some changes in the library. You would define the SoftSPIB Object like this:
````
SoftSPIB mySPI(12, 13, 4);
````
and then pass the mySPI object like this: 
````
ADS1220_WE ads = ADS1220_WE(ADS1220_CS_PIN, ADS1220_DRDY_PIN, &mySPI); 
````
On the library side you would need a constructor that takes the SoftSPIB object, like:
````
ADS1220_WE::ADS1220_WE(int cs, int drdy, SoftSPIB *s){
     _softSPI = s;
````
A variable would need to be defined:
````
SoftSPI *_spi;
````
And then finally all SPI functions (line 324 till the end) in ADS1220_WE.cpp would need to be changed to the corresponding SoftSPIB functions, beauce they are a bit different. 

So in summary: the library could be adjusted, but with some effort.

If may ask why do you want to use SoftSPIB? Why don't you just use the standard SPI interface of the ESP32?

**Comment by wollewald** [comment]
I have just tried an example sketch with an ESP32 Development Board using the standard SPI interface. I only had to adjust the DRDY Pin and the CS Pin. The rest was unchanged.

**Comment by xiyangdiy** [comment]
Thanks for your enthusiastic answer!
I'm just a beginner, your answer helped me a lot!

**Comment by wollewald** [comment]
You are welcome!
